File: app/src/main/java/com/gooner10/popularmoviesapp/data/MovieDao.java
package com.gooner10.popularmoviesapp.data;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

/**
 * MovieDao for Room
 */

@Dao
public interface MovieDao {
    @Query("SELECT * FROM moviedata")
    List<MovieItem> getAllFavoritMovie();

    @Insert
    void insertAll(MovieItem... movieData);

    @Delete
    void deleteMovies(MovieItem movieData);

    @Query("SELECT * FROM moviedata where id LIKE :movieId")
    boolean movieExists(int movieId);
}

File: core/data/src/main/java/com/gooner10/data/MovieDao.kt
package com.gooner10.data

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.gooner10.data.model.MovieItem

/**
 * MovieDao for Room
 */

@Dao
interface MovieDao {
    @get:Query("SELECT * FROM moviedata")
    val allFavoritMovie: List<MovieItem>

    @Insert
    fun insertAll(vararg movieData: MovieItem)

    @Delete
    fun deleteMovies(movieData: MovieItem)

    @Query("SELECT * FROM moviedata where id LIKE :movieId")
    fun movieExists(movieId: Int): Boolean
}
